recipe/recipe.go

```diff
@@ -0,0 +1,214 @@
+package main
+
+import (
+	"archive/zip"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/JulzDiverse/cfclient"
+	"github.com/julz/cube"
+	"github.com/pkg/errors"
+	"github.com/starkandwayne/goutils/ansi"
+
+	"code.cloudfoundry.org/bbs/models"
+	"code.cloudfoundry.org/runtimeschema/cc_messages"
+)
+
+func main() {
+	downloadUrl := os.Getenv(cube.EnvDownloadUrl)
+	uploadUrl := os.Getenv(cube.EnvUploadUrl)
+	appId := os.Getenv(cube.EnvAppId)
+	stagingGuid := os.Getenv(cube.EnvStagingGuid)
+	completionCallback := os.Getenv(cube.EnvCompletionCallback)
+
+	username := os.Getenv(cube.EnvCfUsername)
+	password := os.Getenv(cube.EnvCfPassword)
+	apiAddress := os.Getenv(cube.EnvApiAddress)
+	cubeAddress := os.Getenv(cube.EnvCubeAddress)
+
+	fmt.Println("STARTING WITH:", downloadUrl, uploadUrl, appId, stagingGuid, completionCallback)
+
+	annotation := cc_messages.StagingTaskAnnotation{
+		CompletionCallback: completionCallback,
+	}
+
+	annotationJson, err := json.Marshal(annotation)
+	if err != nil {
+		os.Exit(1)
+	}
+
+	cfclient, err := cfclient.NewClient(&cfclient.Config{
+		SkipSslValidation: true,
+		Username:          username,
+		Password:          password,
+		ApiAddress:        apiAddress,
+	})
+
+	respondWithFailureAndExit(err, stagingGuid, annotationJson)
+
+	downloader := Downloader{cfclient}
+	uploader := Uploader{cfclient}
+
+	err = downloader.Download(appId, "/workspace/appbits")
+	respondWithFailureAndExit(err, stagingGuid, annotationJson)
+
+	// TODO: Replace this with pure-go implementation of an unzipper
+	err = execCmd(
+		"unzip", []string{
+			"/workspace/appbits",
+		})
+	respondWithFailureAndExit(err, stagingGuid, annotationJson)
+
+	err = os.Remove("/workspace/appbits")
+	respondWithFailureAndExit(err, stagingGuid, annotationJson)
+
+	// TODO: Replace this with an inplace call of packs library
```

Comment:
maybe we should set up a tracker or use GitHub issues to track these todos? (then we can use that rather than todo comments)

Reply:
good idea, i will open issues for the todo's.